File: core/src/main/java/com/cangwang/core/cwmodule/ex/CWFacotry.java
package com.cangwang.core.cwmodule.ex;

import com.cangwang.model.ICWModule;
import com.cangwang.model.IModuleFactory;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;

/**
 * Created by yyinc on 2017/12/17.
 */

public class CWFacotry implements IModuleFactory{
    public static Map<String,List<ICWModule>> moduleMap = new HashMap<>();

    static {
        List<ICWModule> list = new LinkedList<>();
//        list.add();
//        moduleMap.put("normal",);
        list.clear();
    }

    @Override
    public List<ICWModule> getTempleList(String templet) {
        return moduleMap.get(templet);
    }
}

File: core/src/main/java/com/cangwang/core/cwmodule/ex/CWFacotry.kt
package com.cangwang.core.cwmodule.ex

import com.cangwang.model.ICWModule
import com.cangwang.model.IModuleFactory
import java.util.*

/**
 * Created by yyinc on 2017/12/17.
 */
class CWFacotry : IModuleFactory {
    companion object {
        var moduleMap: Map<String, List<ICWModule>> = HashMap()

        init {
            val list: List<ICWModule> = LinkedList()
            //        list.add();
//        moduleMap.put("normal",);
        }
    }

    override fun getTempleList(templet: String): List<ICWModule> {
        return moduleMap[templet]!!
    }
}